NewForce-Cohort-7/tabloidfullstack-agnostic-pillows
# Ticket #36 Upload User Profile Image

# Description
This is on the `nm-uploaduserprofileimage` branch.

For your reference: [SQL Data](https://github.com/NewForce-Cohort-7/tabloidfullstack-agnostic-pillows/tree/main/SQL)

**# Title** in Trello - [Ticket #36 Upload User Profile Image](https://trello.com/c/qQqD8f3t/37-36-upload-user-profile-image)
Per the ticket: As a user I would like to be able to upload an avatar image so that I do not have to find an image online to link to.

This functionality has now been implemented. 

## Type of change

# Testing Instructions

**GIT this branch:**
1. Clone this repository to your own machine using `git clone https://github.com/NewForce-Cohort-7/tabloidfullstack-agnostic-pillows.git`
2. `git fetch -a` in the master branch to retrieve this branch named `branch-name`.
3.  checkout this branch using the command `git checkout branch-name`


**Create the database:**
1. Open Visual Studio.
2. At the start screen select "Open a Local Folder"
4. Navigate to your workspace directory, open the `tabloidfullstack-agnostic-pillows` folder, then select the SQL folder
5. Run both .sql files in the [SQL folder](https://github.com/NewForce-Cohort-7/tabloidfullstack-agnostic-pillows/tree/main/SQL) (The first file creates the database and tables if they don’t already exist, and the second file inserts a very small amount of sample data into those tables for testing.).

**Run the app:**
1. Open Visual Studio
2. At the start screen select Open a project or solution.
3. Navigate to your workspace directory, open the `tabloidfullstack-agnostic-pillows` folder, and select the .sln file to run the solution.
4. Hit the Green play button on the toolbar that says TabloidFullStack.
5. This will launch the Swagger page in your default browser. The Swagger page provides the endpoints and data from the SQL server that we created. 
6. Now we need to run the React side of the app. 
7. In your command line, cd into the `client` folder and into the `tabloid` folder. 
8. Start by running `npm install react-scripts`. Once this has been installed, type `npm start`. 
9. This will open the React app in your default browser. You should now see the log-in page. 
10. Log in to user `foo@bar.com`. You do not need to enter a password. 

**Test the new feature:**
1. Navigate to the Users page from the header.
2. Find the user you logged in with, in this case it would be "Foo Barington"
3. Click the name to navigate to the details page. You should see a page of user details, along with a button under the profile picture to Change Profile Avatar. Click this.
4. Once there you will see a file upload form. Choose a picture file from your local machine and click upload.
5. You should now see an updated version of your profile details with the new profile picture displayed.


# Checklist:
